### PR TITLE
839 populate expression profile from pk sim db

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/MoBi.Core/Services/IPKSimStarter.cs [hunk 1]
```diff
@@ -8,5 +10,6 @@ public interface IPKSimStarter
       void StartWithWorkingJournalFile(string journalFilePath);
       IBuildingBlock CreateProfileExpression(ExpressionType expressionType);
       IBuildingBlock CreateIndividual();
+      List<ExpressionParameterValueUpdate> UpdateExpressionProfileFromDatabase(ExpressionProfileBuildingBlock expressionProfile);
```
Comment (msevestre) [inline src/MoBi.Core/Services/IPKSimStarter.cs:13]: IReadonLy?

File: src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs [hunk 2]
```diff
@@ -28,6 +36,35 @@ public IReadOnlyList<ExpressionProfileBuildingBlock> LoadFromPKML()
...
+         macroCommand.AddRange(expressionProfileUpdate.Select(parameter => updateCommandFor(buildingBlock, parameter.Path, parameter.UpdatedValue)));
+
+         return macroCommand.Run(Context);
+      }
+
+      private static ICommand updateCommandFor(ExpressionProfileBuildingBlock buildingBlock, ObjectPath path, double? value)
+      {
+         var parameterToUpdate = buildingBlock.FirstOrDefault(x => Equals(x.Path, path));
+         
+         if (parameterToUpdate == null)
```
Comment (msevestre) [inline src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs:62]: can you verify in the history that you do not get empty command int he parent macro command?

File: src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs [hunk 3]
```diff
@@ -28,6 +36,35 @@ public IReadOnlyList<ExpressionProfileBuildingBlock> LoadFromPKML()
...
+         };
+
+         macroCommand.AddRange(expressionProfileUpdate.Select(parameter => updateCommandFor(buildingBlock, parameter.Path, parameter.UpdatedValue)));
+
+         return macroCommand.Run(Context);
+      }
+
+      private static ICommand updateCommandFor(ExpressionProfileBuildingBlock buildingBlock, ObjectPath path, double? value)
+      {
+         var parameterToUpdate = buildingBlock.FirstOrDefault(x => Equals(x.Path, path));
```
Comment (msevestre) [inline src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs:60]: Don;t we have a wa to search for parameter by path that's not o(n)? if feels like we will loop over this for each parameter of the expression profile.

File: src/MoBi.Presentation/Tasks/PKSimStarter.cs [hunk 4]
```diff
@@ -80,9 +83,15 @@ public IBuildingBlock CreateIndividual()
...
+      public List<ExpressionParameterValueUpdate> UpdateExpressionProfileFromDatabase(ExpressionProfileBuildingBlock expressionProfile)
+      {
+         loadPKSimAssembly();
+         return executeMethod(getMethod(PKSIM_UI_STARTER_EXPRESSION_PROFILE_CREATOR, GET_EXPRESSION_DATABASE_QUERY), new object[] { expressionProfile }) as List<ExpressionParameterValueUpdate>;
+      }
+
+      private object executeMethod(MethodInfo method, object[] parameters = null)
       {
-         return method.Invoke(null, null);
+         return method.Invoke(null, parameters);
```
Comment (msevestre) [inline src/MoBi.Presentation/Tasks/PKSimStarter.cs:94]: lol

**Review by msevestre — CHANGES_REQUESTED** [review]
looks good to me. Just a few suggestions

File: src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs:62]: Yes, I verified empty commands don't appear in the history.

File: src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs (hunk 3 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForExpressionProfileBuildingBlock.cs:60]:
```
var parameterToUpdate = buildingBlock[path];
```
?

File: src/MoBi.Presentation/Tasks/PKSimStarter.cs (hunk 4 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/PKSimStarter.cs:94]: Prior to this feature, we haven't had to send anything over the fence to MoBi. We used to send IShell, but we found out that wasn't necessary.

**Review by msevestre — APPROVED** [review]

**Pull request merged**